lib/ide-haskell.coffee

```diff
@@ -14,23 +14,29 @@ configDefaults =
 
 _isCabalProject = false         # true if cabal project
 _pluginManager = null           # plugin manager
+_disposables = new CompositeDisposable
 
 activate = (state) ->
-  _isCabalProject = isCabalProject()
+  projRoot = getCabalProjectDir()
+  _isCabalProject = (projRoot != null)
   $(window).on 'focus', updateMenu
   return unless _isCabalProject
 
+  # store project root on the atom project, since the utils need it
+  atom.project.cabalProjectRoot = projRoot
```

Comment:
@lierdakil, I don't much like this either.  The problem is that the ghc-mod utility needs the project root as the current working directory, and atom doesn't have a "project root" anymore.  I see in your project that you use the first entry from getPaths(), I considered that as well, but it made me nervous to assume the first path was what I needed.  

How about having another level of namespace on this field?  i.e atom.project.ideHaskell.cabalProjectRoot.  That hopefully makes collisions much less likely.

I could also recompute it each time by scanning for the cabal project file on each invocation, and using that to determine the root.